indirect_value.h

```diff
@@ -46,32 +30,38 @@ class indirect_value : private indirect_value_base<T, C> {
     ptr_ = std::unique_ptr<T, D>(new T(std::forward<Ts>(ts)...), D{});
```

Comment:
Should we be using std::make_unique_default_init here? See: https://en.cppreference.com/w/cpp/memory/unique_ptr/make_unique

Feels like we should avoid raw allocation calls if we can.